### PR TITLE
Replace OCP images with public ones

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* replaces OCP images with those which are publicly available so as to not require to openshift org for base images

/cc @periklis @xperimental

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### This is a generated file from Dockerfile.in ###
 #@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.17)
```
Comment (xperimental) [inline Dockerfile:2]: Do the `follow_tag` declarations need any update? Can we remove them?

File: Dockerfile [hunk 2]
```diff
@@ -21,11 +21,11 @@ COPY ${REMOTE_SOURCE}/bundle bundle
 COPY ${REMOTE_SOURCE}/version version
 COPY ${REMOTE_SOURCE}/.bingo ./.bingo
 ADD ${REMOTE_SOURCE}/Makefile ${REMOTE_SOURCE}/main.go ${REMOTE_SOURCE}/go.mod ${REMOTE_SOURCE}/go.sum ./
-
+USER 0
```
Comment (xperimental) [inline Dockerfile:24]: The new image seems to default to having a user `default` with the working directory in `/opt/app-root/src`. As we do not need the "GOPATH-style" of working directory, does it make sense to update the Dockerfile to use the defaults instead?

File: Dockerfile (hunk 1 above)
Comment (periklis) [reply Dockerfile:2]: The `follow_tag` declarations are only used in downstream builds by brew. However to make upstream<->midstream easy we should keep them or else any the diff between them gets bigger. At least we should keep them in the `Dockerfile.in`

**Review by periklis — COMMENTED** [review]
Looks good to me

File: Dockerfile (hunk 2 above)
Comment (jcantrill) [reply Dockerfile:24]: Maybe.  but as @periklis noted this is generated from Dockerfile.in.  Additional, `REMOTE_SOURCE` is deprecated which means there are additional changes to be made.  Adding the user was the same change I made to the CLO which did not seem to have any adverse affects.

Comment (xperimental) [reply Dockerfile:24]: I realized too late that I made the comments on the generated file and not the source :slightly_smiling_face:

**Pull request closed without merging**